engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/EmailValidator.java

```diff
@@ -70,51 +76,42 @@ public boolean isValid(CharSequence value, ConstraintValidatorContext context) {
 			return false;
 		}
 
-		// if we have a trailing dot in local or domain part we have an invalid email address.
-		// the regular expression match would take care of this, but IDN.toASCII drops the trailing '.'
-		// (imo a bug in the implementation)
-		if ( emailParts[0].endsWith( "." ) || emailParts[1].endsWith( "." ) ) {
+		if ( !matchLocalPart( emailParts[0] ) ) {
 			return false;
 		}
 
-		if ( !matchPart( emailParts[0], localPattern, MAX_LOCAL_PART_LENGTH ) ) {
+		return matchDomain( emailParts[1] );
+	}
+
+	private boolean matchLocalPart(String localPart) {
+		if ( localPart.length() > MAX_LOCAL_PART_LENGTH ) {
 			return false;
 		}
-
-		return matchPart( emailParts[1], domainPattern, MAX_DOMAIN_PART_LENGTH );
+		Matcher matcher = LOCAL_PART_PATTERN.matcher( localPart );
+		return matcher.matches();
 	}
 
-	private boolean matchPart(String part, Pattern pattern, int maxLength) {
+	private boolean matchDomain(String domain) {
+		// if we have a trailing dot the domain part we have an invalid email address.
+		// the regular expression match would take care of this, but IDN.toASCII drops the trailing '.'
+		if ( domain.endsWith( "." ) ) {
+			return false;
+		}
+
 		String asciiString;
 		try {
-			asciiString = toAscii( part );
+			asciiString = IDN.toASCII( domain );
 		}
 		catch (IllegalArgumentException e) {
 			return false;
 		}
 
-		if ( asciiString.length() > maxLength ) {
+		if ( asciiString.length() > MAX_DOMAIN_PART_LENGTH ) {
 			return false;
 		}
 
-		Matcher matcher = pattern.matcher( asciiString );
+		Matcher matcher = DOMAIN_PATTERN.matcher( asciiString );
 		return matcher.matches();
 	}
 
-	private String toAscii(String unicodeString) throws IllegalArgumentException {
-		String asciiString = "";
-		int start = 0;
-		int end = unicodeString.length() <= 63 ? unicodeString.length() : 63;
-		while ( true ) {
-			// IDN.toASCII only supports a max "label" length of 63 characters. Need to chunk the input in these sizes
```

Comment:
The chunking was introduced because the local part was mistakenly passed to IDN.toASCII to be converted from UTF-8 to ASCII.

The local part max length is 64 characters while each label of a domain name (parts separated by dots) must be at most 63 characters long. The chunking was introduced to solve this issue. See https://hibernate.atlassian.net/browse/HV-1005 and https://github.com/hibernate/hibernate-validator/pull/422/files .

The fundamental error was to apply IDN.toASCII to the local part in the first place, it's not supposed to be used on other things than domain names where the 63 characters limit on each label is perfectly legit.

So I did the following:
- allow the UTF-8 range that is now allowed in the local part regexp
- only use IDN.toASCII to validate domain names - and this part does not need chunking anymore